RUM - Browser SDK DevTools tips - Guide creation (#16212)
DataDog/documentation: content/en/real_user_monitoring/guide/devtools-tips.md

---

> title: Tips when using browser's developer tools

Author: 👌

kind: guide
further_reading:
    - link: '/real_user_monitoring/browser'
      tag: 'Documentation'
      text: 'RUM Browser Monitoring'
    - link: '/logs/log_collection/javascript'
      tag: 'Documentation'
      text: 'Browser Log Collection'
---

## Overview

> This guide provides some tips when using the browser's developer tools to debug an application instrumented with the Datadog browser SDKs.

Author: 👌

> ## File and line number mismatch in DevTools console

Author: 👌

The browser SDK instruments the console functions (`console.error`, but also `.log`, `.info` and `.warn`) to collect data about the behavior of the application.

> This leads to inconvenient side-effects, as shown below:

Author: 👌

{{< img src="real_user_monitoring/guide/devtools-tips/issue_console.jpg" alt="Devtools console showing issue about wrong file and line numbers for a console.error statement.">}}

> In the picture above, the `console.error` function is instrumented. As a developer, it is expected to see the real file and line number on which this statement was called. However, instead of displaying `VM505:1`, the console shows `datadog-rum-v4.js:1`.

Author: I left 'incorrectly' out - the console is (sadly) correct.

> ### Add scripts to your browser ignore list

Author: 👌

> Most browsers allow developers to select scripts and add them to an ignore list. Doing so for `datadog-rum-v*.js` and `datadog-logs-v*.js` ensures the DevTools display the proper file and line number.

Author: 👌

> How to use this feature on Google Chrome:

Reviewer: ```suggestion
Below, we describe where to find this feature in Google Chrome as an example.
```

Author: 👌

{{< img src="real_user_monitoring/guide/devtools-tips/script_ignore_list.jpg" alt="How to add script to the ignore list in Google Chrome.">}}

> In the console tab, expand the output from the console statement. Right-click on each script you wish to ignore, and select the option 'add script to ignore list'.

Reviewer: ```suggestion
In the console tab, expand the output from the console statement. Right-click on each script you wish to ignore, and select the option, **Add script to ignore list**.
```

Author: 👌

> Note: The ignore list can be managed in Developer Tools > Settings > Ignore List

Reviewer: ```suggestion
**Note:** The ignore list can be managed in **Developer Tools > Settings > Ignore List**.
```

Author: 👌

> This method works well when using the CDN (sync/async) installation methods. If you are using the NPM package, make sure that you have sourcemaps enabled. Else, SDK code may be bundled with your application code, preventing DevTools to ignore the SDK.

Reviewer: ```suggestion
This method works well when using the [CDN (sync/async) installation methods][3]. If you are using the NPM package method, make sure that you have `sourcemaps` enabled. Otherwise, the SDK code may be bundled with your application code, preventing DevTools from ignoring the SDK.
```

Author: 👌

> Another advantage from using the ignore list is seen in the network panel:

Reviewer: ```suggestion
Another advantage of using the ignore list can be seen in the network panel:
```

Author: 👌

{{< img src="real_user_monitoring/guide/devtools-tips/network_initiator.jpg" alt="Network initiator properly showing after adding scripts to ignore list.">}}

> Instead of displaying the browser SDK as the initiator of the request, the proper file and line number are shown for the application.

Reviewer: ```suggestion
Instead of displaying the browser SDK as the initiator of the request, the correct file and line number are shown for the application.
```

Author: 👌

## Remove noise in the network tab

> The browser SDKs send several network requests to record an application's behavior. This can generate a significant number of lines in the network tab, and makes it hard to spot the requests initiated by your application. Most browsers allow to filter the requests coming from the browser SDKs.

Reviewer: ```suggestion
The browser SDKs send several network requests to record an application's behavior. This can generate a significant number of lines in the network tab, which makes it hard to identify the requests initiated by your application. Most browsers allow you to filter the requests coming from the browser SDKs.
```

Author: 👌

> Below an example of this feature on Google Chrome:

Reviewer: ```suggestion
Below an example of this feature in Google Chrome:
```

Author: 👌

{{< img src="real_user_monitoring/guide/devtools-tips/network_ignore_intake.jpg" alt="Network panel filtering browser SDKs requests.">}}

In the network tab, add a filter of the form `-url:intake-datadoghq.com` (update the pattern to match the url of your [datacenter's intake][1], or the one of your [proxy][2]).

## Further Reading

{{< partial name="whats-next/whats-next.html" >}}

[1]: /getting_started/site

> [2]: /real_user_monitoring/guide/proxy-rum-data

Reviewer: ```suggestion
[2]: /real_user_monitoring/guide/proxy-rum-data
[3]: /real_user_monitoring/browser/#choose-the-right-installation-method
```

Author: 👌

---
title: Tips when using browser's developer tools
kind: guide
further_reading:
    - link: '/real_user_monitoring/explorer'
      tag: 'Documentation'

>       text: 'Visualize your RUM data in the Explorer'

Reviewer 2: <!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏no idea if there is some rules for that but suggesting explorer doc here seems odd to me.
I would say maybe having the general browser RUM doc and the general browser log doc could be nice.

Author: 👌

Another advantage from using the ignore list is seen in the network panel:
{{< img src="real_user_monitoring/guide/devtools-tips/network_initiator.jpg" alt="Network initiator properly showing after adding scripts to ignore list.">}}

Instead of displaying the browser SDK as the initiator of the request, the proper file and line number are shown for the application.

## Remove noise in the network tab

The browser SDKs will send several network requests to record an application's behavior. This can generate a significant number of lines in the network tab, and makes it hard to spot the requests initiated by your application. Most browsers will allow you to filter the requests coming from the browser SDKs.

Below an example of this feature on Google Chrome:
{{< img src="real_user_monitoring/guide/devtools-tips/network_ignore_intake.jpg" alt="Network panel filtering browser SDKs requests.">}}

> In the network tab, add a filter of the form `-url:intake-datadoghq.com` (update the pattern to match the url of your datacenter's intake, or the one of your proxy).

Reviewer 2: <!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it could be nice to add links to the [site doc](https://docs.datadoghq.com/getting_started/site/) and to the [proxy doc](https://docs.datadoghq.com/real_user_monitoring/guide/proxy-rum-data/?tab=npm).

Author: 👌
